### ISSUE TITLE
EFR32 BLEManagerImpl::SendIndication is assuming the passed-in data buffer is "small enough"

### ISSUE THREAD

**Issue by bzbarsky-apple** [project-chip/connectedhomeip#2595]
#### Problem
The code does:
```
    rsp = gecko_cmd_gatt_server_send_characteristic_notification(conId, cId, data->DataLength(), data->Start());
```
which is passing that third (data length) arg as a `uint8_t`.  What happens if it overflows?

 #### Proposed Solution
Check for too-large data length, error out if it's too large.

**Comment by bzbarsky-apple** [comment]
@sagar-apple @vivien-apple @shana-apple

**Comment by jepenven-silabs** [comment]
@bzbarsky-apple will be fixed by the GSDK update from 2.7 to 3.1 (in progress)

**Comment by bzbarsky-apple** [comment]
@jepenven-silabs Sounds good.  When that happens, please remove the cast to `uint8_t` at the callsite described above?
